### PR TITLE
UHM-4935, use the encounterDate datepicker widget value

### PR DESCRIPTION
**Pull request by cioan**
@mogoodrich , here is, I think a safer way to get the encounter date of the form. I am not sure what changed, I am pretty sure the encounter lookup expression used to work for new encounters, either in the past or present. But, anyway this PR addresses this problem. Thanks.

### CODE REVIEW

File: configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml [hunk 1]
```diff
@@ -120,10 +120,21 @@
 
         var currentEncounterDate = new Date();
         var encounterDate = '<lookup expression="encounter.getEncounterDatetime().getTime()"/>';
-        if (typeof encounterDate !== "undefined" &amp;&amp; encounterDate !== null &amp;&amp; (encounterDate.length > 0)) {
-          currentEncounterDate = new Date(+encounterDate);
+          if (typeof encounterDate !== "undefined" &amp;&amp; encounterDate !== null &amp;&amp; (encounterDate.length > 0)) {
+            currentEncounterDate = new Date(+encounterDate);
+        } else {
+          // look for the encounterDate datepicker widget
+          var encounterDateValue = jq("#encounterDate .hasDatepicker");
```
Comment (mogoodrich) [inline configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml:127]: Does the getField method not work here?  (Not a big deal if it doesn't, just curious)

File: configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml [hunk 2]
```diff
@@ -185,11 +196,7 @@
 
 
         var validateNextDispensingDate = function() {
-            var currentEncounterDate = new Date();
-            var encounterDate = '<lookup expression="encounter.getEncounterDatetime().getTime()"/>';
-            if (typeof encounterDate !== "undefined" &amp;&amp; encounterDate !== null &amp;&amp; (encounterDate.length > 0)) {
-                currentEncounterDate = new Date(+encounterDate);
-            }
+            var currentEncounterDate = new Date(getField('encounterDate.value').datepicker('getDate'));
```
Comment (mogoodrich) [inline configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml:199]: Just new Date(getField('encounterDate.value'))" isn't enough?

Again, just curious, not a big deal if it works as-is...

**Review by mogoodrich — APPROVED** [review]

File: configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml (hunk 1 above)
Comment (cioan) [reply configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml:127]: Thanks @mogoodrich ! Yes, I have noticed in the javascript console that sometimes when the form loads it throws a null getField() method not found. I have tried to debug it, but when I set a breakpoint it works b/c there is plenty time for the page and all depending js scripts to load. When I switched to using jquery selector I did not see that error anymore.

Comment (mogoodrich) [reply configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml:127]: Interesting... something to keep an eye on for other places where it may cause issues.

**Pull request merged**